components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java

```diff
@@ -314,21 +316,28 @@ private boolean validateIdToken(String idToken) {
      * There is a problem If Id token signed using SP's tenant and there is no direct way to get the tenant domain
      * using client id. So have iterate all the Tenants until get the right client id.
      * @param idToken id token
-     * @return Tenant domain
+     * @param tenantDomain tenant domain sent in request params.
+     * @return Tenant domain for signature validation.
      */
-    private String getTenantDomainForSignatureValidation(String idToken) {
+    private String getTenantDomainForSignatureValidation(String idToken, String tenantDomain) {
         boolean isJWTSignedWithSPKey = OAuthServerConfiguration.getInstance().isJWTSignedWithSPKey();
-        String tenantDomain;
+        String tenantDomainForSignatureValidation;
 
         try {
             String clientId = extractClientFromIdToken(idToken);
             if (isJWTSignedWithSPKey) {
                 OAuthAppDO oAuthAppDO = OAuth2Util.getAppInformationByClientId(clientId);
-                tenantDomain = OAuth2Util.getTenantDomainOfOauthApp(oAuthAppDO);
+                tenantDomainForSignatureValidation = OAuth2Util.getTenantDomainOfOauthApp(oAuthAppDO);
             } else {
-                //It is not sending tenant domain with the subject in id_token by default, So to work this as
-                //expected, need to enable the option "Use tenant domain in local subject identifier" in SP config
-                tenantDomain = MultitenantUtils.getTenantDomain(extractSubjectFromIdToken(idToken));
+                if (StringUtils.isNotBlank(tenantDomain)) {
```

Comment:
It would be worth to add a few debug logs to indicate which path the request went through.